ext/ImplicitDifferentiationForwardDiffExt.jl

```diff
@@ -39,7 +43,9 @@ function (implicit::ImplicitFunction)(
         reshape(dₖy_vec, size(y))
     end
 
-    y_and_dy = let y = y, dy = dy
+    y_and_dy = if y isa Number
```

Comment:
Is this type-stable or should it be two methods?

Reply:
If the type of `y` is inferred correctly the code will be type stable, if it isn't type stable, you will have a problem either case

example:
```julia
julia> foo(x) = x isa Int ? 2 : 2.0
foo (generic function with 1 method)

julia> @code_warntype foo(3)
MethodInstance for foo(::Int64)
  from foo(x) @ Main REPL[15]:1
Arguments
  #self#::Core.Const(foo)
  x::Int64
Body::Int64
1 ─ %1 = (x isa Main.Int)::Core.Const(true)
└──      goto #3 if not %1
2 ─      return 2
3 ─      Core.Const(:(return 2.0))
```
the interesting part is `(x isa Main.Int)::Core.Const(true)`